OOP/OOP-3/TrainReservation.java

```diff
@@ -0,0 +1,322 @@
+//this class is used to print chart of all the trains
+class TrainChart
+{
+	ArrayList<String> trainInfo=new ArrayList<String>();
+	//Iterator<String> itTrainInfo=trainInfo.iterator();
+	String trainType,trainNumber;
+	private int numberOfSeats;
+	public TrainChart(String trainType,ArrayList<String> trainInfo)
+	{
+		this.trainType=trainType;
+		this.trainInfo=trainInfo;
+	}
+	public void chart()//To show all the trains of either of goods or passangers
+	{
+		System.out.println("TrainId	From	To	S/W	Ticket	Duration");
+		for(int i=0;i<trainInfo.size();i++)
+		System.out.println(trainInfo.get(i));
+	}
+	//this method is used for show the chart of trains after booking
+	public void updatedChart(int numberOfSeats,String trainNumber) 
+	{
+		int i,j;
+		if(trainType.equalsIgnoreCase("goods"))
+		{
+			
+			String temp1[]=new String[6];
+			StringBuffer sb=new StringBuffer();
+			this.numberOfSeats=numberOfSeats;
+			this.trainNumber=trainNumber;
+			for(i=0;i<trainInfo.size();i++)//Loop for subtracting the number of booked seats
+			{
+				
+				temp1=trainInfo.get(i).split(",");
+				if(temp1[0].equalsIgnoreCase(trainNumber)) 
+				{
+					int hold=Integer.parseInt(temp1[3])-numberOfSeats;
+					temp1[3]=Integer.toString(hold);
+					break;
+				}
+			}
+			
+			try
+			{
+				FileWriter fw=new FileWriter("Goods.txt");
+				for(int k=0;k<trainInfo.size();k++)
+				{
+					String temp2=trainInfo.get(k);
+					if(k==i)
+					{
+						for(j=0;j<6;j++)
+						{
+							fw.write(temp1[j]+",");
+						}
+						fw.write("\n");
+					}
+					else
+						fw.write(temp2+"\n");
+				}
+				fw.close();
+				String line;
+				FileReader fr=new FileReader("Goods.txt");
+				BufferedReader bf=new BufferedReader(fr);
+				while((line=bf.readLine())!=null)
+				{
+					System.out.println(line);
+				}
+			}catch(Exception obException){}
+			
+		}
+		else if(trainType.equalsIgnoreCase("Passanger"))
+		{
+			String temp1[]=new String[6];
+			StringBuffer sb=new StringBuffer();
+			this.numberOfSeats=numberOfSeats;
+			this.trainNumber=trainNumber;
+			for(i=0;i<trainInfo.size();i++)
+			{
+				
+				temp1=trainInfo.get(i).split(",");
+				if(temp1[0].equalsIgnoreCase(trainNumber)) 
+				{
+					int hold=Integer.parseInt(temp1[3])-numberOfSeats;
+					temp1[3]=Integer.toString(hold);
+					break;
+				}
+			}
+			try	
+			{
+				FileWriter fw=new FileWriter("Passanger.txt");
+				for(int k=0;k<trainInfo.size();k++)
+				{
+					String temp2=trainInfo.get(k);
+					if(k==i)
+					{
+						for(j=0;j<6;j++)
+						fw.write(temp1[j]+",");
+						fw.write("\n");
+					}
+					else
+						fw.write(temp2+"\n");
+				}
+				fw.close();
+				String line;
+				FileReader fr=new FileReader("Passanger.txt");
+				BufferedReader bf=new BufferedReader(fr);
+				while((line=bf.readLine())!=null)
+					System.out.println(line);
+				} catch(Exception e) {}
+			}
+		
+	}
+}
+//this class is used for checking the availability of required seats 
+class CheckAvailability
```

Comment:
There is no point of creating this extra non-static class, this could have been done in a static method.